gcloud-java-resourcemanager/src/main/java/com/google/gcloud/resourcemanager/testing/LocalResourceManagerHelper.java

```diff
@@ -0,0 +1,593 @@
+    } else {
+      project.setLifecycleState("DELETE_REQUESTED");
+      response = new Response(HttpURLConnection.HTTP_OK, "{}");
+    }
+    return response;
+  }
+
+  Response get(String projectId, String[] fields) {
+    if (!projects.containsKey(checkNotNull(projectId))) {
+      // when possible, change this to 404 (#440)
+      return Error.PERMISSION_DENIED.response("Project " + projectId + " not found.");
+    }
+    Project project = projects.get(projectId);
+    try {
+      return new Response(
+          HttpURLConnection.HTTP_OK, jsonFactory.toString(extractFields(project, fields)));
+    } catch (IOException e) {
+      return Error.INTERNAL_ERROR.response(
+          "Error when serializing project " + project.getProjectId());
+    }
+  }
+
+  Response list(final Map<String, Object> options) {
+    // Use pageSize and pageToken options when Cloud Resource Manager does so (#421)
+    List<String> projectsSerialized = new ArrayList<>();
+    for (Project p : projects.values()) {
+      Boolean includeProject = includeProject(p, (String[]) options.get("filter"));
+      if (includeProject) {
+        try {
+          projectsSerialized.add(
+              jsonFactory.toString(extractFields(p, (String[]) options.get("fields"))));
+        } catch (IOException e) {
+          return Error.INTERNAL_ERROR.response(
+              "Error when serializing project " + p.getProjectId());
+        }
+      } else if (includeProject == null) {
+        return Error.INVALID_ARGUMENT.response("Could not parse the filter.");
+      }
+    }
+    StringBuilder responseBody = new StringBuilder();
+    responseBody.append("{\"projects\": [");
+    responseBody.append(Joiner.on(",").join(projectsSerialized));
+    responseBody.append("]}");
+    return new Response(HttpURLConnection.HTTP_OK, responseBody.toString());
+  }
+
+  private static Boolean includeProject(Project project, String[] filters) {
+    if (filters == null) {
+      return true;
+    }
+    for (String filter : filters) {
+      String[] filterEntry = filter.toLowerCase().split(":");
+      if ("id".equals(filterEntry[0])) {
+        if (!satisfiesFilter(project.getProjectId(), filterEntry[1])) {
+          return false;
+        }
+      } else if ("name".equals(filterEntry[0])) {
+        if (!satisfiesFilter(project.getName(), filterEntry[1])) {
+          return false;
+        }
+      } else if (filterEntry[0].startsWith("labels")) {
+        String labelKey = filterEntry[0].split("\\.")[1];
+        if (project.getLabels() != null) {
+          String labelValue = project.getLabels().get(labelKey);
+          if (!satisfiesFilter(labelValue, filterEntry[1])) {
+            return false;
+          }
+        }
+      } else {
+        return null;
+      }
+    }
+    return true;
+  }
+
+  private static boolean satisfiesFilter(String projectValue, String filterValue) {
+    if (projectValue == null) {
+      return false;
```

Comment:
Yes, according to the [public documentation](https://cloud.google.com/resource-manager/reference/rest/v1beta1/projects/list).  So for example `name:*` only returns projects that have a non-null name.